policybot/handlers/githubwebhook/filters/testresultfilter/testresultfilter.go

```diff
@@ -23,65 +23,56 @@ import (
 	"istio.io/bots/policybot/handlers/githubwebhook/filters"
 	"istio.io/bots/policybot/pkg/config"
 	gatherer "istio.io/bots/policybot/pkg/resultgatherer"
-	"istio.io/bots/policybot/pkg/storage"
 	"istio.io/bots/policybot/pkg/storage/cache"
 	"istio.io/pkg/log"
 )
 
 // Updates the DB based on incoming GitHub webhook events.
-type ResultGatherer struct {
-	store              storage.Store
-	repos              map[string]bool
-	cache              *cache.Cache
-	testResultGatherer *gatherer.TestResultGatherer
+type TestResultFilter struct {
+	repos map[string]gatherer.TestResultGatherer
+	cache *cache.Cache
+	// testResultGatherer *gatherer.TestResultGatherer
 }
 
 var scope = log.RegisterScope("ResultGatherer", "Result gatherer for each pr test run", 0)
 
-func NewResultGatherer(store storage.Store,
-	cache *cache.Cache, orgs []config.Org, bucketName string) filters.Filter {
+func NewTestResultFilter(cache *cache.Cache, orgs []config.Org, bucketName string) filters.Filter {
 	ctx := context.Background()
 
 	client, err := s.NewClient(ctx)
```

Comment:
Instead of creating the client here, please take a blob store as a parameter.  The blob store is created in cmd/server.go already. That client is an abstraction over GCS and uses the right credentials.